java/hiro-client/src/main/java/co/arago/hiro/client/rest/DefaultHiroClient.java

```diff
@@ -136,6 +127,20 @@ public DefaultHiroClient(String restApiUrl, TokenProvider tokenProvider, AsyncHt
 
     }
 
+    private static void checkError(Response r) throws HiroException {
+        if (r.hasResponseStatus() && r.getStatusCode() >= 400) {
+            if (r.hasResponseBody()) {
+                final Map info = Helper.parseJsonBody(r.getResponseBody());
+                Map error = (Map) info.get("error");
```

Comment:
it might be unsafe but would work with current state of things

Reply:
What did I miss that makes it unsafe?

Reply:
what if "error" would be not a map